backend/src/services/secret-sync/aws-parameter-store/aws-parameter-store-sync-fns.ts

```diff
@@ -132,34 +286,75 @@ const deleteParametersBatch = async (
 
 export const AwsParameterStoreSyncFns = {
   syncSecrets: async (secretSync: TAwsParameterStoreSyncWithCredentials, secretMap: TSecretMap) => {
-    const { destinationConfig } = secretSync;
+    const { destinationConfig, syncOptions } = secretSync;
 
     const ssm = await getSSM(secretSync);
 
-    // TODO(scott): KMS Key ID, Tags
-
     const awsParameterStoreSecretsRecord = await getParametersByPath(ssm, destinationConfig.path);
 
+    const awsParameterStoreMetadataRecord = await getParameterMetadataByPath(ssm, destinationConfig.path);
+
+    const awsParameterStoreTagsRecord = await getParameterStoreTagsRecord(
+      ssm,
+      destinationConfig.path,
+      awsParameterStoreSecretsRecord,
+      Boolean(syncOptions.tags || syncOptions.syncSecretMetadataAsTags)
+    );
+
+    const syncTagsRecord = Object.fromEntries(syncOptions.tags?.map((tag) => [tag.key, tag.value]) ?? []);
+
     for await (const entry of Object.entries(secretMap)) {
-      const [key, { value }] = entry;
+      const [key, { value, secretMetadata }] = entry;
 
-      // skip empty values (not allowed by AWS) or secrets that haven't changed
-      if (!value || (key in awsParameterStoreSecretsRecord && awsParameterStoreSecretsRecord[key].Value === value)) {
+      // skip empty values (not allowed by AWS)
+      if (!value) {
         // eslint-disable-next-line no-continue
         continue;
       }
 
-      try {
-        await putParameter(ssm, {
-          Name: `${destinationConfig.path}${key}`,
-          Type: "SecureString",
-          Value: value,
-          Overwrite: true
+      // create parameter or update if changed
+      if (
+        !(key in awsParameterStoreSecretsRecord) ||
+        value !== awsParameterStoreSecretsRecord[key].Value ||
+        syncOptions.keyId !== awsParameterStoreMetadataRecord[key]?.KeyId
```

Comment:
this will always trigger an update when keyId is not explicitly configured by the user because `syncOptions.keyId` would be undefined while `awsParameterStoreMetadataRecord[key]?.KeyId` would be the default KMS key